lib/brakeman/checks/check_reverse_tabnabbing.rb

```diff
@@ -0,0 +1,43 @@
+require 'brakeman/checks/base_check'
+
+class Brakeman::CheckReverseTabnabbing < Brakeman::BaseCheck
+  Brakeman::Checks.add_optional self
+
+  @description = "Checks for reverse tabnabbing cases on 'link_to' calls"
+
+  def run_check
+    calls = tracker.find_call :methods => :link_to
+    calls.each do |call|
+      process_result call
+    end
+  end
+
+  def process_result result
+    return unless original? result and result[:call].third_arg
+
+    html_opts = result[:call].third_arg
+    _, target = hash_access html_opts, :target
+    _, rel = hash_access html_opts, :rel
+    return unless target
+
+    target_url = result[:call].second_arg
+
+    # `_url` and `_path` lead to urls on to the same origin.
+    # That means that an adversary would need to run javascript on
+    # the victim application's domain. If that is the case, the adversary
+    # already has the ability to redirect the victim user anywhere.
+    if target_url[0] == :call then
+      func_s = target_url.method.to_s
+      return unless !func_s.end_with?("url") && !func_s.end_with?("path")
```

Comment:
Not sure I agree with checking for `url`. That could easily be an attacker-controllable value on a model.

In my version I did `url.method.match?(/^url_for$|_path$/)` (although I haven't checked for false positives yet).

Reply:
You are right, and I've also tested this in conjuction with [this](https://github.com/rails/rails/issues/29893). And links that use the `_url` helper are vulnerable. If the the `_url` helper method is used (the same is true for `url_for` calls), the `X-Forwarded-Host` header can be used to poison the cache and so control the domain of the resulting `<a>` tags for as long as the cache holds the cache entry.

So you are specifically looking for `url_for`/`_path` or you are ruling those out?

Reply:
I think `url_for`/`_path` should be ignored. The issue with `X-Forwarded-Host` aside, these are most often going to be used for internal links.